File: project_euler/problem_3.c
#include <limits.h>
#include <stdio.h>

int main () {
    long long int large_factor=0, i=2;
    long long int number=600851475143;

    printf("long long int max : %lld\n",LLONG_MAX);

    while (number!=1) {
        if (number%i==0) {
            while (number%i==0)
                number/=i;
            large_factor=i;
        }
        i++;
     }

    printf ("\n\nThe largest prime factor is : %lld\n\n", large_factor);

    return 0;
}

File: project_euler/problem_3.rb
def largest_prime( max )
    largest = 0, i = 2

    while 1 != max do
        if 0 == max % i then
            while 0 == max % i do
                max /= i;
            end
            largest = i
        end
        i += 1
    end

    return largest
end

puts largest_prime( 600851475143 );
